src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -442,13 +435,10 @@ private static ModuleMetadata ReadModuleMetadataFrom(ObjectReader reader, Serial
         var array = reader.ReadByteArray();
         var pinnedObject = new PinnedObject(array);
 
-        var metadata = ModuleMetadata.CreateFromMetadata(pinnedObject.GetPointer(), array.Length);
-
-        // make sure we keep storageStream alive while Metadata is alive
-        // we use conditional weak table since we can't control metadata liftetime
-        s_lifetimeMap.Add(metadata, pinnedObject);
-
-        return metadata;
+        // PinnedObject will be kept alive as long as the ModuleMetadata is alive due to passing its .Dispose method in
+        // as the onDispose callback of the metadata.
+        return ModuleMetadata.CreateFromMetadata(
+            pinnedObject.GetPointer(), array.Length, pinnedObject.Dispose);
```

Comment:
the overload that took an onDispose callback didn't exist at the time when this code was originally written, leading to having to have the CWT that would keep the object alive.  This is no longer necessary as the callback itself can keep it alive.